### PR TITLE
Bitmap Preparation

### PR DESCRIPTION
**Pull request by sidschingis**
- Added shortcut function for rolling a random number.
- Changed Enums to bitflags
    - Requires a menu update
-Added util functions for bitmap handling

### CODE REVIEW

File: src/common/consts.rs [hunk 1]
```diff
@@ -20,81 +49,85 @@ pub enum HitboxVisualization {
...
     match direction {
         Direction::None => ANGLE_NONE,
         Direction::Random => ANGLE_NONE, // Random Direction should be handled by the calling context
-        _ => (direction as i32 - 1) as f64 * PI / 4.0,
+        // Translate to bit position using trailing_zeros first
+        _ => (direction as u32).trailing_zeros() as f64 * PI / 4.0,
     }
 }
 
+pub unsafe fn get_direction(bitmap: BitMap) -> Direction {
```
Comment (jugeeya) [inline src/common/consts.rs:94]: should probably be named `get_random_direction`? Also, is this used?

File: src/common/consts.rs [hunk 2]
```diff
@@ -20,81 +49,85 @@ pub enum HitboxVisualization {
...
+    UpRight = 0x2,
+    Up = 0x4,
+    UpLeft = 0x8,
+    Left = 0x10,
+    DownLeft = 0x20,
+    Down = 0x40,
+    DownRight = 0x80,
     // lol what goes here jug smh my head
-    Random = 9,
+    Random = 0x100,
```
Comment (jugeeya) [inline src/common/consts.rs:61]: Just a suggestion, maybe use bit shifting so this code is easier to read? 
```rust
pub enum Direction {
    None = 0,
    Right = 1 << 0,
    UpRight = 1 << 1,
    Up = 1 << 2,
    UpLeft = 1 << 3,
    Left = 1 << 4,
    DownLeft = 1 << 5,
    Down = 1 << 6,
    DownRight = 1 << 7,
    // lol what goes here jug smh my head
    Random = 1 << 8,
}
```

Comment (sidschingis) [reply src/common/consts.rs:61]: Does it work with pattern matching?

File: src/common/consts.rs (hunk 1 above)
Comment (sidschingis) [reply src/common/consts.rs:94]: None of these are used yet. I wanted to have the internal representation updated first before actually applying bitmaps. Still need to get comfortable with the tesla side

File: src/common/consts.rs (hunk 2 above)
Comment (jugeeya) [reply src/common/consts.rs:61]: Ah, it won't, good point.

**Pull request closed without merging**